pkg/controller/quota/quota.go

```diff
@@ -161,6 +162,12 @@ func addStorage(appInstance *v1.AppInstance, quotaRequest *adminv1.QuotaRequestI
 			size = boundSize
 		}
 
+		// If the volume will be bound to an existing PV implicitly, then we don't need to count it. This will happen
+		// when the incoming app has the same name as the app that created an existing released PV.
+		if pvName, err := appdefinition.LookupExistingPV(req, appInstance, name); err == nil && pvName != "" {
+			continue
+		}
```

Comment:
If `LookupExistingPV` errors, would you want to error here? I am guessing that if `LookupExistingPV` returns an error, then we aren't able to determine if we should count the volume again the quota, so may want to retry?

Reply:
You bring up a good point. I did this intentionally but looking back, the current behavior is wrong. Good catch.